Basic-Car-Maintenance/Shared/BasicCarMaintenanceApp.swift

```diff
@@ -8,20 +8,14 @@
 import FirebaseCore
 import SwiftUI
 
-class AppDelegate: NSObject, UIApplicationDelegate {
-    func application(_ application: UIApplication,
-                     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
-    ) -> Bool {
-        FirebaseApp.configure()
-        return true
-    }
-}
-
 @main
 struct BasicCarMaintenanceApp: App {
-    @UIApplicationDelegateAdaptor(AppDelegate.self) var delegate
+    @State private var authenticationViewModel: AuthenticationViewModel
     
-    @StateObject var authenticationViewModel = AuthenticationViewModel()
+    init() {
+        FirebaseApp.configure()
+        self._authenticationViewModel = .init(initialValue: AuthenticationViewModel())
+    }
```

Comment:
The new SwiftUI lifecycle which uses the init method. It is basically the same as the didFinishLaunching in the old lifecycle. But I can revert it if you want

Reply:
Can you link documentation that says this is the same thing?
Because previously it wasn't, so I wanted to double check, also because the AppDelegate has many other things it does, I'm confused how all of that now is just in `init`

Reply:
Here is the part in the [Apple documentation](https://developer.apple.com/documentation/swiftui/uiapplicationdelegateadaptor): Important hint in the middle

Here is an explanation from Peter Friese [how you can initialise Firebase](https://stackoverflow.com/a/62633158)

But if you want I can switch it back to the "old" way with the AppDelegate.

Reply:
That's what I wanted to see thanks!